hack/check-EOF.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ROOTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)"
+
+CMD="/usr/bin/find $ROOTDIR/"
+
+EXCLUDE_DIR=(
+    vendor
+    output
+    _tools
+    .git
+    static/tidb-operator-overview.png
+)
+
+for DIR in ${EXCLUDE_DIR[@]}; do
+    CMD="$CMD -path $ROOTDIR/$DIR -prune -o"
+done
+
+FILELIST=$($CMD -type f -print)
+
+check() {
+    x=$(tail -c 1 "$1")
+    if [ "$x" != "" ]; then
+        echo "$1"
+    fi
+}
+
+NUM=0
+for f in ${FILELIST[@]}; do
+    c=$(tail -c 1 "$f")
+    if [ "$c" != "" ]; then
```

Comment:
this check is false-positive when the last character is a space, you can use `wc -l` to check if it is a newline

```
c=$(tail -c 1 "$f" | wc -l)
if [ $c -gt 0 ]; then
   // newline found
fi
```